From 57f4c82e68953529d5365b0dac736d494d2f6229 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 30 Jan 2019 15:25:14 +0200
Subject: [PATCH] [SW-2807] fw_upd: switch magic numbers to enums

We are about to extend the non-reset flops commands, so to prepare
switch to use enum instead of magic numbers.

Change-Id: I6a2acf567a46ec016660ca65af3e9c982fabe890
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot.c b/preboot.c
index 55b44d0..238e4c7 100644
--- a/preboot.c
+++ b/preboot.c
@@ -10,6 +10,7 @@
 #include <linux/io.h>
 #include <linux/sizes.h>
 #include <mach/hl_regs.h>
+#include <mach/hl_fw_if.h>
 #include "auto_generated.h"
 #include <stddef.h>
 #include <stdbool.h>
@@ -180,16 +181,16 @@
 	char *spl_image = "candidate";
 
 	modifiable_image_pending = readl(mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
-	if (modifiable_image_pending == 3) {
+	if (modifiable_image_pending == FLOPS_UPD_PENDING) {
 		next_image_modifiable = (*(unsigned int*)(HL_SPI_BASE + 0x60004) & 0xff) - '1';
 		if (next_image_modifiable < 2) {
 			next_image = next_image_modifiable;
-			writel((next_image + 1), mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
+			writel((next_image + FLOPS_UPD_NXT_IMG1), mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
 			try_new_image = true;
 		} else
-			writel(0, mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
+			writel(FLOPS_UPD_NOOP, mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
 	} else if (modifiable_image_pending) /* ONESHOT */
-		writel(0, mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
+		writel(FLOPS_UPD_NOOP, mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
 
 	if (!try_new_image) {
 		spl_image = "stable";
